app/code/Magento/CatalogUrlRewrite/Model/ResourceModel/Category/GetDefaultUrlKey.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogUrlRewrite\Model\ResourceModel\Category;
+
+use Magento\Catalog\Api\Data\CategoryInterface;
+use Magento\Catalog\Model\Category;
+use Magento\Eav\Model\Config as EavConfig;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Select;
+use Magento\Framework\EntityManager\MetadataPool;
+use Magento\Store\Model\Store;
+
+/**
+ * Fetch category 'url_key' default value from the database.
+ */
+class GetDefaultUrlKey
+{
+    /**
+     * @var MetadataPool
+     */
+    private $metadataPool;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var EavConfig
+     */
+    private $eavConfig;
+
+    /**
+     * @param MetadataPool $metadataPool
+     * @param ResourceConnection $resourceConnection
+     * @param EavConfig $eavConfig
+     */
+    public function __construct(
+        MetadataPool $metadataPool,
+        ResourceConnection $resourceConnection,
+        EavConfig $eavConfig
+    ) {
+        $this->metadataPool = $metadataPool;
+        $this->resourceConnection = $resourceConnection;
+        $this->eavConfig = $eavConfig;
+    }
+
+    /**
+     * Retrieve 'url_key' value for default store.
+     *
+     * @param int $categoryId
+     * @return string|bool
```

Comment:
I would be better to return string|null

Reply:
Fixed

Reply:
Then we always need add check for a null, which is not good, i think it will be better return always string as per strict type, and in case of empty result empty string. What do you think ?